app/services/common_platform/email_validator.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+module CommonPlatform
+  class EmailValidator < ApplicationService
+    def initialize(email:)
+      @email = email
+    end
+
+    def call
+      JSON::Validator.validate(schema, email, validate_schema: true)
+    end
+
+    private
+
+    attr_reader :email
+
+    def schema
+      {
```

Comment:
**not a blocker** just a a suggestion to consider: is there any way to tie together the swagger json and the schema snippet here (to use snippet in swagger) - so that there not two sources for these regex patterns

Reply:
I have been considering adding the HMCTS schemas itself as maybe a gem/some sort of dependency between both the mock and CDA.
Using the schema snippet as well makes a lot of sense. I will make an investigation around this.

Reply:
Added a [ticket](https://dsdmoj.atlassian.net/browse/CACP-308)